libs/galley-types/src/Galley/Types/Teams.hs

```diff
@@ -236,6 +247,35 @@ data Perm =
     -- read Note [team roles] first.
     deriving (Eq, Ord, Show, Enum, Bounded)
 
+data Role = RoleOwner | RoleAdmin | RoleMember | RoleCollaborator
+    deriving (Eq, Ord, Show, Enum, Bounded)
+
+rolePermissions :: Role -> Permissions
+rolePermissions role = Permissions p p  where p = rolePerms role
+
+rolePerms :: Role -> Set Perm
+rolePerms RoleOwner = rolePerms RoleAdmin <> Set.fromList
```

Comment:
If we want to introduce roles now, we should make sure that role `Owner` = `fullPermissions` or something like this; I am afraid that otherwise we may very well add a new permission and forget to add it (types won't help us there)

Reply:
since you already approved: yes, but different PR!  :-)

(won't be a visible change outside of the wire services.  also need to think about the options for a minute.)